solution.rb

```diff
@@ -152,51 +125,26 @@
             ]
         }
       ]
-      
-      
-      coffee_average_per_person = []
-    
-    # Q_3
-    # ```
-    #   #######The solution by using .each loop#######
-    # people.each do |hash|  ##first loop for hash 
-    #     new_person = {name:"",coffee_average:0}
-    #     new_person[:name] = hash[:name]
-    #     puts hash[:name]
-    #     counter = 0
-    #     total_amout = 0
-    #     hash[:transactions].each do |key,value|  ##second loop for transcations [{},{},{}]
-    #         if key[:type] == "COFFEE"
-    #             total_amout += key[:amount]
-    #             counter += 1
-    #         end
+      avrage = 0
+      sum = 0
+    people.each do |name|
+      names =  name[:name]
```

Comment:
Great work; but you need to declare ```names``` array before ```people.each``` and then inside  ```people.each``` you push every name into ```names``` like this:
```rb
names = []
people.each do |name|
  names =  names.push(name[:name])
end
```